[WPE] Use WPE Platform API by default in MiniBrowser and layout tests

carlosgcampos opened the pull request:
#### cf657a695cf4bde0281989ada5562e21201e76d5
<pre>
[WPE] Use WPE Platform API by default in MiniBrowser and layout tests
<a href="https://bugs.webkit.org/show_bug.cgi?id=293354">https://bugs.webkit.org/show_bug.cgi?id=293354</a>

Reviewed by Adrian Perez de Castro.

Change the options to enable the new api into enabling legacy api.

Canonical link: <a href="https://commits.webkit.org/295604@main">https://commits.webkit.org/295604@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/bb7be66fdf542c80dd598287ff4873c242c8e5a3

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/105617 "1 style error") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/25366 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/15758 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/110810 "Built successfully") | [  ~~🛠 win~~](https://ews-build.webkit.org/#/builders/59/builds/56220 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/107658 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/25824 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/33867 "Built successfully") | [❌ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/80207 "Found 1 new test failure: imported/w3c/web-platform-tests/css/css-view-transitions/transform-origin-view-transition-group.html (failure)") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/59/builds/56220 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/108623 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/20168 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/95311 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/60516 "Passed tests") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/105095 "Passed tests") | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/133/builds/19908 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/13403 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/55648 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/89618 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/13443 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/113654 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/32755 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/24143 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/89284 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/33117 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/91540 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/88948 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/33821 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/11621 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/28196 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/17126 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/32681 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/38065 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/32427 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/35775 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/34024 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

spenap reviewed (commented):
Looks good, thorough job inverting all checks!

aperezdc reviewed (approved):
It's exciting that we are reaching the point where we can do the switch to WPEPlatform 👏🏼!

csaavedra commented on Tools/WebKitTestRunner/libwpe/EventSenderProxyLibWPE.cpp:
```diff
@@ -48,7 +48,7 @@ EventSenderProxy::EventSenderProxy(TestController* testController)
+    if (!testController->useWPELegacyAPI())
         m_client = makeUnique<EventSenderProxyClientWPE>(*testController);
     else
         m_client = makeUnique<EventSenderProxyClientLibWPE>(*testController);
```
While this diff keeps the change to a minimum, I think it is better to avoid negated clauses when it's possible  (you can swap the if / else blocks).

csaavedra commented on Tools/WebKitTestRunner/libwpe/PlatformWebViewLibWPE.cpp:
```diff
@@ -43,7 +43,7 @@ PlatformWebView::PlatformWebView(WKPageConfigurationRef configuration, const Tes
+    if (!TestController::singleton().useWPELegacyAPI())
         m_window = new PlatformWebViewClientWPE(configuration);
     else
         m_window = new PlatformWebViewClientLibWPE(configuration);
```
Ditto.

csaavedra reviewed (commented):
Some style comments.

carlosgcampos committed:
[WPE] Use WPE Platform API by default in MiniBrowser and layout tests
https://bugs.webkit.org/show_bug.cgi?id=293354

Reviewed by Adrian Perez de Castro.

Change the options to enable the new api into enabling legacy api.

Canonical link: https://commits.webkit.org/295604@main
